src/extensions/nginx-app-protect/monitoring/processor/nap.go

```diff
@@ -624,6 +547,32 @@ func (f *NAPConfig) getViolations(logger *logrus.Entry) []*models.ViolationData
 	return violations
 }
 
+func populateNameValue(violationName, dataName, dataValue string) (name string, value string) {
+	if dataName != "" && dataValue != "" {
+		name = dataName
+		value = dataValue
+		return
+	}
+	name = violationNameToDataName(violationName)
+	if dataName == "" && dataValue != "" {
+		value = dataValue
+		return
+	}
+	if dataName != "" && dataValue == "" {
+		value = dataName
+		return
+	}
+	return
+}
+
+func violationNameToDataName(violationName string) string {
+	parts := strings.Split(strings.ToLower(violationName), string(violationNameSeparator))
+	if parts[0] != violationNamePrefix {
+		return strings.Join(parts, " ")
+	}
+	return strings.Join(parts[1:], " ")
```

Comment:
Should we convert these to title case? They would be slightly neater on the frontend imo.

```suggestion
	return strings.Title(strings.Join(parts[1:], " "))
```
The above only works if the violation name is in `strings.ToLower()` first.